src/process/loader-shell-runner.ts

```diff
@@ -1,36 +1,30 @@
 import { ShellOutput } from "./shell-output.ts";
 import { bold, brightBlue } from "../deps.ts";
 import { IShellRunner } from "./shell-runner.interface.ts";
-import * as tty from "../commands/tty.ts";
+import { TTY } from "../commands/tty.ts";
 
 export class LoaderShellRunner implements IShellRunner {
   private interval?: number;
   private runningCommandsCount = 0;
 
-  private isDisabled: boolean;
   private sigInt: Deno.SignalStream | null = null;
   private sigTerm: Deno.SignalStream | null = null;
 
-  constructor(private runner: IShellRunner) {
-    // we must determine this once because later checks can throw an error if we
-    // call this too often while logging due to race conditions:
-    // error: Uncaught (in promise) Busy: Resource is unavailable because it is in use by a promise
-    this.isDisabled = !Deno.isatty(Deno.stdout.rid);
+  constructor(
+    private runner: IShellRunner,
+    private readonly tty: TTY,
```

Comment:
Ah well, it only ends up being injected into the Shellrunners. Guess that's not too bad! :-)

Reply:
Yeah the design choice was not super critical. imo classes are always superior because of injectability and thus better testability. but since we don't test much and injecting mocks is a PITA in TS I don't know if this helps a lot... but on the other hand it also does not hurt. :D